### ISSUE TITLE
sudo: sorry, you are not allowed to set the following environment variables

### ISSUE THREAD

**Issue by krathalan** [archlinux/devtools#76]
I have been running makechrootpkg as an unprivileged non-sudo-enabled user for a long time with the following `/etc/sudoers` snippet:

```
builder ALL=(ALL) NOPASSWD:/usr/bin/makechrootpkg
```

After installing the `20211129-1` devtools package, I now get the following error running `makechrootpkg`:

```
sudo: sorry, you are not allowed to set the following environment variables: BUILDTOOL, BUILDTOOLVER
```

Here is the full `bash -x makechrootpkg` output: https://gist.github.com/krathalan/d5103a058c210b672241ad874987860e

**Comment by heftig** [comment]
```
<heftig> i can't use makechrootpkg on build.archlinux.org anymore
<heftig> makechrootpkg -r /var/lib/archbuild/extra-x86_64
<heftig> sudo: sorry, you are not allowed to set the following environment variables: BUILDTOOL, BUILDTOOLVER
<heftig> i don't think BUILDTOOL and BUILDTOOLVER should be passed to check_root
<heftig> they'll get reset when the tool gets reexecuted, anyway
```

/cc @anthraxx

**Comment by anthraxx** [comment]
This is true, but mostly just because there isn't really a difference with the current implementation for BUILDTOOL as we forcefully set it via common.sh. Theoretically it may be set conditionally in the future to allow wrapper overrides.

All the env variables that may be passed along can lead to this error, if you start exporting used variables like GNUPGHOME,SRCDEST,SRCPKGDEST,PKGDEST,LOGDEST etc you will face the same issue. Otherwise, when not checking, they are unexpectedly overwritten (not preserved).

If you want to run makechrootpkg like this from a sudoers context, one needs to preserve the env variables that are implemented as a feature in the sudoers file via `env_keep+=`

**Comment by heftig** [comment]
> If you want to run makechrootpkg like this from a sudoers context, one needs to preserve the env variables that are implemented as a feature in the sudoers file via env_keep+=

But if you don't set any environment variables, you shouldn't need to modify sudoers, either.

> Theoretically it may be set conditionally in the future to allow wrapper overrides.

Can't we remove them from preservation now? We can still preserve them when that future happens. That the tool now sets its own variables it expects to get preserved needlessly makes this a regression.

**Comment by foutrelis** [comment]
> Can't we remove them from preservation now?

+1 (I think archbuild doesn't preserve them either.)

**Comment by anthraxx** [comment]
We should rather adjust BUILDTOOL to be set conditionally instead of 
force overwrite as we would need that anyway if we support downstream 
tooling overrides. This environment variable would be set by default 
similar to how SOURCE_DATE_EPOCH is set by default in archbuild and also 
handled by our infrastructure deployment exactly out of the same reason:

https://gitlab.archlinux.org/archlinux/infrastructure/-/blob/master/roles/archbuild/files/sudoers#L7

https://gitlab.archlinux.org/archlinux/infrastructure/-/commit/6560a2dfc0cea4d834e38e572e3d011184607a46
